### PR TITLE
merge Project and View

### LINKED ISSUE #746: move Classloadingoptions, Language, etc. logic into AnalysisInputLocation
why - e.g.
- the `Language` / Java version is mainly needed for the multi release jar
- ClassLoadingOption is stored as a function which gets the analysisinputlocation as parameter.. so in the end its a map like container depending on AnalysisInputLocation
- AnalysisInputLocation for apks: platform jar options

### PR DESCRIPTION
**Pull request by stschott**
Changes:
- Move Language into MultiReleaseJarInputLocation
- Move List of BodyInterceptors to apply to AnalysisInputLocation
- Merge Project and View
- Remove Scope

closes #746

### CODE REVIEW

File: sootup.java.bytecode/src/main/java/sootup/java/bytecode/inputlocation/ModuleFinder.java [hunk 1]
```diff
@@ -210,6 +210,7 @@ private void discoverModulesIn(@Nonnull Path path) {
 
   private void buildModuleForExplodedModule(@Nonnull Path dir) throws ResolveException {
     // create the input location for this module dir
+    // TODO: propagte corresponding BodyInterceptors to newly cerated InputLocations
```
Comment (JonasKlauke) [inline sootup.java.bytecode/src/main/java/sootup/java/bytecode/inputlocation/ModuleFinder.java:213]: did you create an issue for this?

**Review by JonasKlauke — APPROVED** [review]

File: sootup.java.bytecode/src/main/java/sootup/java/bytecode/inputlocation/JavaClassPathAnalysisInputLocation.java [hunk 2]
```diff
@@ -69,6 +72,11 @@ public JavaClassPathAnalysisInputLocation(@Nonnull String classPath) {
     this(classPath, SourceType.Application);
   }
 
+  public JavaClassPathAnalysisInputLocation(
+      @Nonnull String classPath, @Nonnull SourceType srcType) {
+    this(classPath, srcType, new ArrayList<>());
```
Comment (swissiety) [inline sootup.java.bytecode/src/main/java/sootup/java/bytecode/inputlocation/JavaClassPathAnalysisInputLocation.java]:
```suggestion
    this(classPath, srcType, Collections.emptyList());
```
// + import

**Comment by JonasKlauke** [comment]
could you test getClassSources on MultiReleaseJarAnalysisInputLocation. it is completly untested

**Comment by swissiety** [comment]
> could you test getClassSources on MultiReleaseJarAnalysisInputLocation. it is completly untested

I already added a test in #738

File: sootup.java.bytecode/src/main/java/sootup/java/bytecode/inputlocation/JavaModulePathAnalysisInputLocation.java [hunk 3]
```diff
@@ -68,14 +68,19 @@ public class JavaModulePathAnalysisInputLocation implements ModuleInfoAnalysisIn
...
   }
 
   public JavaModulePathAnalysisInputLocation(
       @Nonnull String modulePath, @Nonnull SourceType sourcetype) {
     this(modulePath, FileSystems.getDefault(), sourcetype);
   }
 
+  public JavaModulePathAnalysisInputLocation(
+      @Nonnull String modulePath, @Nonnull FileSystem fileSystem, @Nonnull SourceType sourcetype) {
+    this(modulePath, fileSystem, sourcetype, new ArrayList<>());
```
Comment (swissiety) [inline sootup.java.bytecode/src/main/java/sootup/java/bytecode/inputlocation/JavaModulePathAnalysisInputLocation.java:81]:
```suggestion
    this(modulePath, fileSystem, sourceType, Collections.emptyList());
```
// + import

**Review by swissiety — APPROVED** [review]

File: sootup.tests/src/main/java/sootup/tests/SimpleSootClient.java [hunk 4]
```diff
@@ -48,10 +46,8 @@ public static void main(String[] args) {
     AnalysisInputLocation<JavaSootClass> walaSource =
         new JavaSourcePathAnalysisInputLocation(Collections.singleton(javaSourcePath));
 
-    JavaProject p = JavaProject.builder(new JavaLanguage(8)).addInputLocation(walaSource).build();
-
     // 1. simple case
-    JavaView fullView = p.createView();
+    JavaView fullView = new JavaView(walaSource);
```
Comment (swissiety) [inline sootup.tests/src/main/java/sootup/tests/SimpleSootClient.java]: deleeete me, deleeeeete my Bod...Class! Remove this class.

**Pull request merged**